File: kirun-java/src/test/java/com/fincity/nocode/kirun/engine/json/schema/convertor/NullConvertorTest.java
package com.fincity.nocode.kirun.engine.json.schema.convertor;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import java.util.Collections;

import org.junit.jupiter.api.Test;

import com.fincity.nocode.kirun.engine.json.schema.Schema;
import com.fincity.nocode.kirun.engine.json.schema.convertor.enums.ConversionMode;
import com.fincity.nocode.kirun.engine.json.schema.convertor.exception.SchemaConversionException;
import com.google.gson.JsonElement;
import com.google.gson.JsonNull;
import com.google.gson.JsonPrimitive;

class NullConvertorTest {

	private JsonElement convertElement(ConversionMode mode, JsonElement element) {
		return NullConvertor.convert(Collections.emptyList(), new Schema(), mode, element);
	}

	@Test
	void testNullConvertNullElement() {
		JsonElement result = convertElement(ConversionMode.STRICT, null);
		assertEquals(JsonNull.INSTANCE, result);
	}

	@Test
	void testNullConvertNullJsonElement() {
		JsonElement result = convertElement(ConversionMode.STRICT, JsonNull.INSTANCE);
		assertEquals(JsonNull.INSTANCE, result);
	}

	@Test
	void testNullConvertStringNull() {
		JsonElement element = new JsonPrimitive("null");
		JsonElement result = convertElement(ConversionMode.STRICT, element);
		assertEquals(JsonNull.INSTANCE, result);
	}

	@Test
	void testNullConvertInvalidElement() {
		JsonElement element = new JsonPrimitive("invalid");
		assertThrows(SchemaConversionException.class, () -> convertElement(ConversionMode.STRICT, element));
	}

	@Test
	void testNullConvertLenientMode() {
		JsonElement element = new JsonPrimitive("invalid");
		JsonElement result = convertElement(ConversionMode.LENIENT, element);
		assertEquals(JsonNull.INSTANCE, result);
	}

	@Test
	void testNullConvertUseDefaultMode() {
		JsonElement element = new JsonPrimitive("invalid");
		JsonElement result = convertElement(ConversionMode.USE_DEFAULT, element);
		assertEquals(JsonNull.INSTANCE, result);
	}

	@Test
	void testNullConvertSkipMode() {
		JsonElement element = new JsonPrimitive("invalid");
		JsonElement result = convertElement(ConversionMode.SKIP, element);
		assertEquals(element, result);
	}

}

File: kirun-js/__tests__/engine/json/schema/covnertor/NullConvertorTest.ts
import { ConversionMode } from '../../../../../src/engine/json/schema/convertor/enums/ConversionMode';
import { Schema } from '../../../../../src';
import { NullConvertor } from '../../../../../src/engine/json/schema/convertor/NullConvertor';
import { SchemaConversionException } from '../../../../../src/engine/json/schema/convertor/exception/SchemaConversionException';

const convertElement = (mode: ConversionMode, element: any) => {
    return NullConvertor.convert([], new Schema(), mode, element);
};

test('Null convert null element (strict mode)', async () => {
    const result = convertElement(ConversionMode.STRICT, null);
    expect(result).toBe(null);
});

test('Null convert null JSON element (strict mode)', async () => {
    const result = convertElement(ConversionMode.STRICT, null);
    expect(result).toBe(null);
});

test('Null convert string "null" (strict mode)', async () => {
    const element = 'null';
    const result = convertElement(ConversionMode.STRICT, element);
    expect(result).toBe(null);
});

test('Null convert invalid element (strict mode)', async () => {
    const element = 'invalid';
    expect(() => convertElement(ConversionMode.STRICT, element)).toThrow(SchemaConversionException);
});

test('Null convert invalid element (lenient mode)', async () => {
    const element = 'invalid';
    const result = convertElement(ConversionMode.LENIENT, element);
    expect(result).toBe(null);
});

test('Null convert use default mode', async () => {
    const element = 'invalid';
    const result = convertElement(ConversionMode.USE_DEFAULT, element);
    expect(result).toBe(null);
});

test('Null convert skip mode', async () => {
    const element = 'invalid';
    const result = convertElement(ConversionMode.SKIP, element);
    expect(result).toBe(element);
});
